fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoanTransactionsApiResource.java

```diff
@@ -439,6 +477,59 @@ private String retrieveTransaction(final Long loanId, final String loanExternalI
         return this.toApiJsonSerializer.serialize(settings, transactionData, this.responseDataParameters);
     }
 
+    private Page<LoanTransactionData> retrieveTransactions(final Long loanId, final String loanExternalIdStr, final String excludedTypesStr,
+            Pageable pageable, final UriInfo uriInfo) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        ExternalId loanExternalId = ExternalIdFactory.produce(loanExternalIdStr);
+        Long resolvedLoanId = getResolvedLoanId(loanId, loanExternalId);
```

Comment:
We can skip this and immediately fetching transaction by loan external id, no?

Reply:
This also checks if the loan exists at all, and throws a LoanNotFoundException if not.

Reply:
Except this way you have a different behaviour if loan id was provided and external id was provided:
- If API called with loan id there is no check whether the loan is existing or not... it will simply return 0 transactions
- If API called with external id, there is a check which check whether the loan is existing or not....

Reply:
Ohh you're right, i just glanced over getResolvedLoanId earlier and didn't realise it only checked with external id. Fixed it.